DOCS-10936 NetSuite 11.2 (#1469)
mulesoft/docs-connectors: netsuite/11.2/modules/ROOT/pages/netsuite-connector-upgrade-migrate.adoc

= Upgrade to NetSuite Connector 11.x
:page-aliases: connectors::netsuite/netsuite-connector-upgrade-migrate.adoc

Upgrade NetSuite Connector to version 11.x to use the new features.

== Supported Upgrade Paths

[%header,cols="50a,50a"]
|===
|From Version | To Version
|10.x |11.x
|===

== Changes in This Release

> In the NetSuite Connector 10.x, the connector resolved metadata by receiving Java objects as input and converting them internally to XML so that NetSuite could understand them. In NetSuite Connector 11.x, XML metadata is resolved directly.

Reviewer: ```suggestion
In NetSuite Connector 10.x, the connector resolved metadata by receiving Java objects as input, and converting those objects internally to XML, so that NetSuite could understand it. In NetSuite Connector 11.x, XML metadata is resolved directly without the need for conversion.
```

Reviewer: "them" is wrong because we are talking about NetSuite understanding the metadata.

Because of this modification, the input and output metadata for connector operations has changed and is now equivalent to the request and response bodies of the corresponding SOAP API operations. You can use DataSense explorer in Studio to drag and drop the metadata fields, which generates their corresponding namespaces and tags automatically.

Version 11.x also eliminates the REST-based operations.

[[changed-sources]]
== Changes to Sources

> Removed the Page Size field for the sources (On Deleted Object, On Modified Object, and On New Object). Each source now returns one record in the output metadata.

Reviewer: ```suggestion
Removed the *Page Size* field for the sources *On Deleted Object*, *On Modified Object*, and *On New Object*. Each source now returns one record in the output metadata.
```

| Async Get List | Removed the Retrieved Base Ref Types field. You now specify the associated values in the Message field of the input metadata.
| Async Get List Records | Removed this operation. For equivalent functionality, use the Get List operation and specify `FILE` as the record type in the Message field of the input metadata.
| Async Initialize List | Removed the Initialize Records field. You now specify the associated values in the Message field of the input metadata.
| Async Search | Removed the Search Record and Criteria fields. You now specify the associated values in the Message field of the input metadata.
| Async Update List | Renamed Record Type to Type. Removed Records attributes. You now specify the associated values in the Message field of the input metadata.
| Async Upsert List | Renamed Record Type to Type. Removed Records attributes. You now specify the associated values in the Message field of the input metadata.
| Check Async Status | No changes.
| Get Async Result | No changes.
|===

> === Invitee operations

Reviewer: ```suggestion
=== Invitee Operations
```
